### PR TITLE
Performance/canary testing with SLO validation tutorials

### PR DESCRIPTION
**Pull request by sriumcp**
# This PR is now replaced by #2752. The latter has the same content has this one and also resolves the git commit issue.
***

**What this PR does / why we need it**:
This PR provides documentation samples for simple and easy performance testing with service-level objectives (SLOs) for KServe models. The samples cover both HTTP and gRPC-based inferencing.

This PR also provides documentation samples for canary testing using Iter8 for Kserve, using Prometheus metrics.

**Type of changes**
Please delete options that are not relevant.

- [x] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

We ran the entire tutorials end-to-end multiple times. The tutorials are fully self-contained, so all the configuration needed to run them are described as part of the tutorials themselves.

**Special notes for your reviewer**:

This is purely a documentation sample. The Iter8 team had earlier contributed the metrics & monitoring section of the Kserve documentation. Part of this PR builds on this metrics and monitoring testing to do performance validation with service-level objectives (SLOs).

### CODE REVIEW

**Comment by yuzisun** [comment]
@sriumcp Can you help fix the git commits, it included the older commits from kserve.

File: docs/samples/canary-testing/README.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+EOF
+```
+
+Verify that your inference service is ready.
+
+```shell
+kubectl wait --for=condition=Ready --timeout=600s isvc/sklearn-iris
+```
+
+## Install Iter8 CLI
```
Comment (yuzisun) [inline docs/samples/canary-testing/README.md:64]: So from what i read from the tutorial, iter8 is now a client tool and no longer requires installing manifest in kubernetes, right?

File: docs/samples/performance-testing/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+> Performance testing of KServe inference services using gRPC is described [here](grpc.md). Canary testing using Prometheus metrics is described [here](../canary-testing/README.md). This tutorial focuses on performance testing of KServe inference services with HTTP endpoints. The main steps in this tutorial are:
+> 1. [Deploy an InferenceService](#deploy-an-inferenceservice)
+> 2. [Launch an Iter8 experiment](#launch-an-iter8-experiment)
+> 3. [View experiment report](#view-experiment-report)
+
+***
+
+## Deploy an InferenceService
+
+Create an InferenceService that exposes an HTTP port. The following serves the SciKit [irisv2 model](https://kserve.github.io/website/0.10/modelserving/v1beta1/sklearn/v2/#deploy-with-inferenceservice):
```
Comment (yuzisun) [inline docs/samples/performance-testing/README.md:18]:
```suggestion
Create an InferenceService that exposes an HTTP port. The following serves the Scikit-learn [irisv2 model](https://kserve.github.io/website/0.10/modelserving/v1beta1/sklearn/v2/#deploy-with-inferenceservice):
```

File: docs/samples/performance-testing/grpc.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+--set ready.timeout=600s \
+--set grpc.protoURL=https://raw.githubusercontent.com/kserve/kserve/master/docs/predict-api/v2/grpc_predict_v2.proto \
+--set grpc.host=sklearn-irisv2-predictor-default.default.svc.cluster.local:80 \
+--set grpc.call=inference.GRPCInferenceService.ModelInfer \
+--set grpc.dataURL=https://gist.githubusercontent.com/kalantar/6e9eaa03cad8f4e86b20eeb712efef45/raw/56496ed5fa9078b8c9cdad590d275ab93beaaee4/sklearn-irisv2-input-grpc.json \
+--set grpc.warmupNumRequests=10 \
+--set assess.SLOs.upper.grpc/latency/mean=500 \
+--set assess.SLOs.upper.grpc/latency/p90=900 \
+--set assess.SLOs.upper.grpc/error-count=0 \
+--set runner=job
```
Comment (yuzisun) [inline docs/samples/performance-testing/grpc.md:71]: Does this launch a kubernetes job to generate the request load?

File: docs/samples/canary-testing/README.md (hunk 1 above)
Comment (sriumcp) [reply docs/samples/canary-testing/README.md:64]: Yes, this is indeed the case for the tutorials you see. To be more exact, there are different capabilities in Iter8. The samples this PR is providing requires only the CLI (which internally interacts with K8s cluster to start Iter8 jobs or cronjobs and collect experiment results).

We will be following up with more PRs were we will showcase how other advanced Iter8 capabilities (like metrics collection for A/B testing experiments) can benefit Kserve. This requires a cluster install of Iter8 service.

File: docs/samples/performance-testing/grpc.md (hunk 3 above)
Comment (sriumcp) [reply docs/samples/performance-testing/grpc.md:71]: Yes.

**Pull request closed without merging**